design/one-pager-consuming-k8s-clusters.md

```diff
@@ -193,78 +212,80 @@ infrastructure owner (i.e. "clusters as cattle").
    `Secret`.
 1. Application owner says that they need to run a sensitive workload on that
    cluster. Infrastructure owner creates a new namespace called
-   `super-secret-workloads` and creates a `WorkloadTarget` in that namespace
+   `super-secret-workloads` and creates a `KubernetesTarget` in that namespace
    that references the previously created Kubernetes `Provider`.
 1. Application owner provisions a `KubernetesApplication` with labels that
-   select that `WorkloadTarget` for provisioning.
+   select that `KubernetesTarget` for provisioning.
 
 This example shows the power of this model, but also one of the notable
 shortcomings: if a user can publish clusters to a namespace with a
-`WorkloadTarget`, then they can consume any cluster. For the application owners
-that are doing self-service provisioning in this scenario, they will likely need
-to be able to create `WorkloadTarget` resources for their dynamically
-provisioned Kubernetes cluster managed resources. However, if they can do so,
-they could just as easily create a `WorkloadTarget` for the Kubernetes
-`Provider` that is only to be used for sensitive workloads. While it would be
-nice to restrict the clusters that could be published by users with the ability
-to create `WorkloadTarget` resources, there is not an immediately clear and
-elegant way to do so. It could be argued that only trusted users should have
-this permission level anyway, so it may be less of an issue than expressed here.
-Nevertheless, a potential solution is outlined [below](#future-considerations).
+`KubernetesTarget`, then they can consume any cluster. For the application
+owners that are doing self-service provisioning in this scenario, they will
+likely need to be able to create `KubernetesTarget` resources for their
```

Comment:
I guess if we keep the ability to provision `KubernetesApplication` to `KubernetesCluster`, those self-serving users wouldn't need to have the role that is needed to create `KubernetesTarget`.

Reply:
@muvaf yep! That way they could consume clusters that they provisioned themselves, but would not be able to enable the consumption of any arbitrary cluster.